server/src/main/java/org/elasticsearch/cluster/metadata/MappingMetadata.java

```diff
@@ -72,6 +77,7 @@ public MappingMetadata(String type, Map<String, Object> mapping) {
             withoutType = (Map<String, Object>) mapping.get(type);
         }
         this.routingRequired = routingRequired(withoutType);
+        this.fieldsForModels = null; // TODO: Set fieldsForModels here
```

Comment:
How would we go about setting `fieldsForModels` in this constructor?

Reply:
Same as previous